src/org/labkey/targetedms/folderImport/PanoramaQCSettings.java

```diff
@@ -0,0 +1,533 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public enum PanoramaQCSettings
+{
+    METRIC_CONFIG (TargetedMSSchema.TABLE_QC_METRIC_CONFIGURATION, QCFolderConstants.QC_METRIC_CONFIGURATION_FILE_NAME)
+            {
+                @Override
+                public void exportSettings(VirtualFile vf, Container container, User user) throws Exception
+                {
+                    TableInfo ti = getTableInfo(user, container, null);
+                    List<ColumnInfo> userEditableCols = ti.getColumns().stream().filter(ci -> ci.isUserEditable()).collect(Collectors.toList());
+                    SimpleFilter filter = SimpleFilter.createContainerFilter(container); //only export the ones that are defined in current container (and not the ones from the root container)
+
+                    Results results = QueryService.get().select(ti, userEditableCols, filter, null);
+                    ResultsFactory factory = ()-> results;
+                    exportSettingsToTSV(vf, factory, getSettingsFileName(), getTableName());
+                    results.close();
+                }
+            },
+    QC_ENABLED_METRICS (TargetedMSSchema.TABLE_QC_ENABLED_METRICS, QCFolderConstants.QC_ENABLED_METRICS_FILE_NAME)
+            {
+                @Override
+                public void exportSettings(VirtualFile vf, Container container, User user) throws Exception
+                {
+                    TargetedMSSchema schema = new TargetedMSSchema(user, container);
+                    TableInfo ti = getTableInfo(user, container, null);
+                    SQLFragment sql = new SQLFragment("SELECT qcMetricConfig.Name AS metric, qcEnabledMetrics.enabled, qcEnabledMetrics.lowerBound, qcEnabledMetrics.upperBound, qcEnabledMetrics.cusumLimit FROM ")
+                            .append(ti, "qcEnabledMetrics")
+                            .append(" INNER JOIN ")
+                            .append(TargetedMSManager.getTableInfoQCMetricConfiguration(), "qcMetricConfig")
+                            .append(" ON qcEnabledMetrics.metric = qcMetricConfig.Id");
+
+                    Results results = QueryService.get().selectResults(schema, sql.getSQL(), null, null, true, true);
+                    ResultsFactory factory = ()-> results;
+                    exportSettingsToTSV(vf, factory, getSettingsFileName(), getTableName());
+                    results.close();
+                }
+
+                @Override
+                public int importSettingsFromFile(FolderImportContext ctx, VirtualFile panoramaQCDir, @Nullable TargetedMSSchema schema, @Nullable TableInfo ti, @Nullable QueryUpdateService qus, @Nullable BatchValidationException errors) throws Exception
+                {
+                    DataLoader loader = new TabLoader(Readers.getReader(panoramaQCDir.getInputStream(getSettingsFileName())), true);
```

Comment:
This and other TabLoader instances need to be closed, ideally in a try-with-resources block.

Reply:
Done! Thanks, Josh!